diff --git a/dist/cjs/index.js b/dist/cjs/index.js
index ab586ec5b7aa9ad84700041d81d6820db144f2e8..fabdd360b73d001a46b4cebd235bf13bf6e520dd 100644
--- a/dist/cjs/index.js
+++ b/dist/cjs/index.js
@@ -991,7 +991,7 @@ var Injector = class Injector {
 		const item = dependency[1];
 		if (this.resolvedDependencyCollection.has(identifierOrCtor)) throw new AddDependencyAfterResolutionError(identifierOrCtor);
 		if (typeof item === "undefined") this.dependencyCollection.add(identifierOrCtor);
-		else if (isAsyncDependencyItem(item) || isClassDependencyItem(item) || isValueDependencyItem(item) || isFactoryDependencyItem(item)) this.dependencyCollection.add(identifierOrCtor, item);
+		else if (isAsyncDependencyItem(item) || isClassDependencyItem(item) || isValueDependencyItem(item) || isFactoryDependencyItem(item) || isExistingDependencyItem(item)) this.dependencyCollection.add(identifierOrCtor, item);
 		else this.resolvedDependencyCollection.add(identifierOrCtor, item);
 	}
 	/**
diff --git a/dist/esm/index.js b/dist/esm/index.js
index e8c58e6de45cc1edff2c9f358bc3f283f9e2c693..e9a35e2ea25e68344e043fb711078886f8ba904c 100644
--- a/dist/esm/index.js
+++ b/dist/esm/index.js
@@ -990,7 +990,7 @@ var Injector = class Injector {
 		const item = dependency[1];
 		if (this.resolvedDependencyCollection.has(identifierOrCtor)) throw new AddDependencyAfterResolutionError(identifierOrCtor);
 		if (typeof item === "undefined") this.dependencyCollection.add(identifierOrCtor);
-		else if (isAsyncDependencyItem(item) || isClassDependencyItem(item) || isValueDependencyItem(item) || isFactoryDependencyItem(item)) this.dependencyCollection.add(identifierOrCtor, item);
+		else if (isAsyncDependencyItem(item) || isClassDependencyItem(item) || isValueDependencyItem(item) || isFactoryDependencyItem(item) || isExistingDependencyItem(item)) this.dependencyCollection.add(identifierOrCtor, item);
 		else this.resolvedDependencyCollection.add(identifierOrCtor, item);
 	}
 	/**
diff --git a/dist/umd/index.js b/dist/umd/index.js
index 7b7076e3e8e9d26aa85f6d7f0f2b3203a6ac79ae..05f91dc713e4715b1a0a553ed33be9bb48209179 100644
--- a/dist/umd/index.js
+++ b/dist/umd/index.js
@@ -996,7 +996,7 @@ var Injector = class Injector {
 		const item = dependency[1];
 		if (this.resolvedDependencyCollection.has(identifierOrCtor)) throw new AddDependencyAfterResolutionError(identifierOrCtor);
 		if (typeof item === "undefined") this.dependencyCollection.add(identifierOrCtor);
-		else if (isAsyncDependencyItem(item) || isClassDependencyItem(item) || isValueDependencyItem(item) || isFactoryDependencyItem(item)) this.dependencyCollection.add(identifierOrCtor, item);
+		else if (isAsyncDependencyItem(item) || isClassDependencyItem(item) || isValueDependencyItem(item) || isFactoryDependencyItem(item) || isExistingDependencyItem(item)) this.dependencyCollection.add(identifierOrCtor, item);
 		else this.resolvedDependencyCollection.add(identifierOrCtor, item);
 	}
 	/**
